magnetic field of straight line

Priyanka-43 opened the pull request:
In this pull request, I have written documentation part of Magnetic Field of wire and try to elaborate with figure as well.

Priyanka-43 committed:
Magnetic Field of Straight wire line

First draft of Magnetic field of wire.

Priyanka-43 committed:
Update Magnetic Field of Straight wire wire line

Priyanka-43 committed:
Update and rename Magnetic Field of Straight wire wire line to MagneticFieldofStraightwire.md

Trying to correct some mistake related to formula and image upload.

kylemasc917 reviewed (changes requested):
This documentation file is complete and well written, all equations are legible latex code.
Just a few things need to be addressed.
1. A description of the materials in the pull request needs to be added.
2. Are the files for the rest of the homework assignment going to be added in a later pull request?
                    -If so, this can be merged once 1. is addressed.
                    -If not, then those files need to be added as soon as possible.

laserlab commented:
> This documentation file is complete and well written, all equations are legible latex code. Just a few things need to be addressed.
> 
> 1. A description of the materials in the pull request needs to be added.
> 2. Are the files for the rest of the homework assignment going to be added in a later pull request?
>    -If so, this can be merged once 1. is addressed.
>    -If not, then those files need to be added as soon as possible.

Agreed. 
@kylemasc917 do you want to also review #46 since there is actual code to review? You don’t have to you’ve already done your duty.

Priyanka-43 commented:
@kylemasc917 
Hi, Can you please explain little more for your first comment? 

And i have made another Pull request for Code.

kylemasc917 commented:
Yeah. The Pull request currently has no description provided (the box at the top is blank). In order for this to be merged according to the homework assignment there needs to be a brief description of the contents of the pull request. You can do this by pressing the three dots in the top right of the first box, pressing edit, and adding a description.

laserlab commented:
> @kylemasc917 Hi, Can you please explain little more for your first comment?
> 
> And i have made another Pull request for Code.

Can you link that PR here? There is no other authored by you in the open PR list in this repository

laserlab commented:
@Priyanka-43 As per homework description only one PR is to be made. If you want to code count to your homework you have to add the files to this PR here NOW.

Priyanka-43 committed:
Create Code_MagneticFieldOfWire.py

I have written code for Magnetic Field of wire. And i have used Lambda function and Generator function.

Priyanka-43 commented:
Ok, I have added my code part to the same pull request.

laserlab commented:
> Ok, I have added my code part to the same pull request.

@kylemasc917 Would you mind reviewing the code as well?

kylemasc917 reviewed (changes requested):
Above issues have been resolved and code runs well and is extremely readable.

Only further changes are that in order to complete the homework assignment there needs to be a unit test module that accompanies the code which is not currently present.

Priyanka-43 committed:
Created using Colab

Priyanka-43 commented:
@kylemasc917 
Hello, I wanted to ask about the unit test part. Shall we used pytest an unit test libraries to make sure it will be running or if it is already giving the results so I am done with the unit test part?

kylemasc917 commented:
Yes you need to write a unit test module even if the code is running properly as a check like constructing a proof for a problem even if you got the expected value.

Priyanka-43 committed:
Update MagneticFieldofStraightwire.md

Added documentation part for Code.
Write code and their corresponding results.

Priyanka-43 committed:
Update MagneticFieldofStraightwire.md

Trying to fix some error.

Priyanka-43 committed:
Update MagneticFieldofStraightwire.md

Fixed some error related to images and did final Commit.

kylemasc917 commented:
A few things. This looks like it is written in Jupyter notebook which will not work as was mentioned in class on Friday I believe, and you never import the function from your code module into the unit test, so it isn't testing anything. If I'm misreading this code feel free to tell me so but those are my observations.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+#Python code for Magnetic field of wire
+import math
```
all import should be at the top of the file or at least grouped together

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+#Python code for Magnetic field of wire Using Generator function
+
+
+import numpy as np
```
multiple import of the same module

Priyanka-43 replied:
Changed into one.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+
+
+import numpy as np
+import matplotlib.pyplot as plt
```
see above and also first comment

Priyanka-43 replied:
Yes. Changed.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+mu_0 = 4 * np.pi * 10**-7  # Permeability of free space (T m/A)
+
+# Parameters
+current = 10  # Current in amperes
```
reuse of variable without prior use

Priyanka-43 replied:
I have changed into one time use of variable.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+distances = np.array([0.01, 0.1, 0.2, 0.3, 0.4, 0.5, 100])  # Distances from 0.01 m to 100 m
+
+# Lambda function to calculate magnetic field
+magnetic_field = lambda I, r: (mu_0 * I) / (2 * np.pi * r)
```
Strongly discourage use of lambda function as discussed in class

Priyanka-43 replied:
Sure.

laserlab replied:
Same as before

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+fields = list(map(lambda r: magnetic_field(current, r), distances))
+
+# Plotting
+plt.figure(figsize=(10, 6))
```
that is not intended behavior when importing a module. Give it a try and import your module and see if that is what you wanted it to do.

Priyanka-43 replied:
Sure.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+plt.grid(True)
+plt.ylim(0, max(fields) * 1.1)
+plt.xlim(0, 0.5)  # Limit x-axis to 0.5 to focus on the initial part of the curve
+plt.show()
```
plots in particular with the show function are much better suited for a jupyter notebook and should not be in a module in most cases.

Priyanka-43 replied:
Ok, Professor, I  will take care of that.

laserlab commented on Code_MagneticFieldOfWire.py:
```diff
@@ -0,0 +1,103 @@
+def magnetic_field_generator(current, distances):
+    """Generator function to yield magnetic field values for given distances."""
+    for r in distances:
+        yield (mu_0 * current) / (2 * np.pi * r)
```
np.pi and math.pi used in the same module for unclear reasons

Priyanka-43 replied:
I have changes in math.pi

laserlab commented on MagneticFieldofStraightwire.md:
```diff
@@ -0,0 +1,53 @@
+It is understood that a moving charge or current generates a magnetic field. A long straight wire carrying a current I is 
+shown in diagram below. The presence of current in the wire creates a magnetic field around it which consist of concentric
+circles. 
+![Screenshot 2024-10-13 164423](https://github.com/user-attachments/assets/25840c22-e22b-402f-86fb-426b548127b3)
```
If you do not have the copyright to this image please remove. If you have it needs a reference.

Priyanka-43 replied:
Hello, I have made this picture by myself.

laserlab reviewed (changes requested):
See inline comments. Also this does not use your electrodynamic function inside a function.

Priyanka-43 committed:
Created using Colab

Priyanka-43 committed:
Update Code_MagneticFieldOfWire.py

Make required changes, that has been suggested by reviewer

laserlab reviewed (changes requested):
- plenty of issues left no merge

Priyanka-43 commented:
Hello Professor, I have tried to resolve. I have done all import in starting of the code. i have deleted the parameters(variables) which I have used more than one. Try to take one module math.pi or np.pi.
